pymedphys/_experimental/pseudonymisation/__init__.py

```diff
@@ -11,12 +11,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
 import functools
 import json
 import logging
 from os.path import abspath, basename, dirname, isdir, isfile
 from os.path import join as pjoin
 
+from pymedphys._imports import immutables
+
+frozenmap = immutables.Map
```

Comment:
Unfortunately, as useful as this is, for now I think it may be more trouble that it is worth. `pymedphys._imports` uses apipkg under the hood (https://pypi.org/project/apipkg/).

The key is, when an attribute on `immutables` is accessed that's when the lazy import "magic" is activated. I don't like using magic... but in this case, the benefit of optional imports justifies the magic I believe...

So, essentially, I imagine in the "clean" test, you'll now be getting an error at this line, as it is trying to import `immutables` when the attribute is accessed.

Reply:
I agree the optional import justifies the magic.
But learning magic is harder than learning physics, so... iterating on the incantations